tests/unit/model-test.js

```diff
@@ -164,6 +164,82 @@ test("isDirty on the individual property is reset after save", function(assert){
     assert.equal(undefined, brandon.get("firstNameIsDirty"));
 });
 
+test("isDirty on the model is reset only after all values set back to original values", function(assert){
+    brandon = Person.create(data);
+    assert.equal("Brandon", brandon.get("firstName"));
+    assert.equal("Williams", brandon.get("lastName"));
+    assert.equal(false, brandon.get("isDirty"));
+    brandon.set("firstName", "foo");
+    brandon.set("lastName", "bar");
+    assert.equal(true, brandon.get("isDirty"));
+    brandon.set("firstName", "Brandon");
+    assert.equal(true, brandon.get("isDirty"));
+    brandon.set("lastName", "Williams");
+    assert.equal(false, brandon.get("isDirty"));
+});
+
+test("isDirty on the model is reset after value set back to original value", function(assert){
+    brandon = Person.create(data);
+    assert.equal("Brandon", brandon.get("firstName"));
+    assert.equal(false, brandon.get("isDirty"));
+    brandon.set("firstName", "baz");
+    assert.equal(true, brandon.get("isDirty"));
+    brandon.set("firstName", "Brandon");
+    assert.equal(false, brandon.get("isDirty"));
+});
+
+test("isDirty on the model is reset when original value is empty string and set back to empty string", function(assert){
+    brandon = Person.create({id: 1, firstName: "", lastName: "Williams"});
+    assert.equal(undefined, brandon.get("firstName"));
+    assert.equal(false, brandon.get("isDirty"));
```

Comment:
This is the only test that feels incorrect/wrong/broken. We start by setting firstName to "" but when you do the getter of firstName it returns undefined. If I do the same thing with ember object it returns firstName as "" (instead of undefined like we have here). This should block this PR for now - I'll give it a look this weekend

Also the final assertion (line 198) seems broken because of this behavior. when reset it shows isDirty as True